azure/scripts/Add-PostgreSQLFirewallRule.ps1

```diff
@@ -0,0 +1,89 @@
+<#
+    .SYNOPSIS
+    Add one or more IP addresses to an Azure PostgreSQL server firewall whitelist
+
+    .DESCRIPTION
+    Add one or more IP addresses to an Azure PostgreSQL server firewall whitelist.
+    If none is provided then use the IP of the current process by making a request to
+    a web service which returns json that includes the request IP.
+
+    .PARAMETER ResourceGroupName
+    The resource group where the Azure PostgreSQL server resides
+
+    .PARAMETER ServerName
+    The Azure PostgreSQL server name
+
+    .PARAMETER FirewallRuleName
+    The name given to the firewall rule
+
+    .PARAMETER IpAddresses
+    Array of IP addresses
+
+    .EXAMPLE
+    $TestParams = @{
+        ResourceGroupName = "ServerResourceGroup"
+        ServerName = $ServerName
+        FirewallRuleName = "agent-ip"
+    }
+    ./Add-PostgreSQLFirewallRule.ps1 @TestParams
```

Comment:
PS C:\> `Get-help .\Add-PostgreSQLFirewallRule.ps1 -Detailed`

@timabell this is so so so much better than your wacky bash script